pkg/registry/core/pod/storage/eviction.go

```diff
@@ -158,23 +158,33 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 			return nil
 		}
 
-		pdb := &pdbs[0]
-		pdbName = pdb.Name
+		var pdb *policyv1beta1.PodDisruptionBudget
 		refresh := false
 		err = retry.RetryOnConflict(EvictionsRetry, func() error {
```

Comment:
Do we need to retry all of the PDBs? I think we should do this inside the loop.

Reply:
Any conflict error from participating PDBs is retried.

Reply:
So my concern here is that the call to `checkAndDecrement` is not idempotent. So if we have 2 pdbs and we get a conflict while updating the last one, we will end up running `checkAndDecrement` twice on the first one. This can lead to situations where a disruption is rejected when it should be allowed.

Would it be possible for us to check in `checkAndDecrement` whether the given pod is already in the `DisruptedPods` map and therefore has already been counted in `PodDisruptionsAllowed`? In this case, multiple calls to `checkAndDecrement` would be less of a concern.

Reply:
Thanks, @mortent for the suggestion.
I have adopted in the latest push.